### PR TITLE
Windows build fixes

### PR DESCRIPTION
**Pull request by michael-loki**

### CODE REVIEW

File: cmake/win32.cmake [hunk 1]
```diff
@@ -24,7 +24,7 @@ if(NOT MSVC_VERSION)
   add_definitions(-DWINVER=0x0500 -D_WIN32_WINNT=0x0500)
   # Wait a minute, if we're not Microsoft C++, nor a Clang paired with Microsoft C++,
   # then the only possible option has to be GNU or a GNU-linked Clang!
-  set(FS_LIB stdc++fs)
+  # set(FS_LIB stdc++fs)
```
Comment (despair86) [inline cmake/win32.cmake]: but why tho

Comment (michael-loki) [reply cmake/win32.cmake]: Latest mingw ships with a non-broken stdlib

Comment (despair86) [reply cmake/win32.cmake]: ok apparently only mainstream targets get this special treatment

t. builds on a bunch of niche, arcane, yet up to date platforms that still ship a separate c++17 filesystem ts lib

Comment (despair86) [reply cmake/win32.cmake]: wrap it around a library check or version check at least

**Comment by despair86** [comment]
Merging, and reversing the patch listed above
thank you for your service mike

**Pull request closed without merging**